lib/trip.rb

```diff
@@ -8,14 +8,68 @@ def initialize(input)
       @id = input[:id]
       @driver = input[:driver]
       @passenger = input[:passenger]
-      @start_time = input[:start_time]
-      @end_time = input[:end_time]
-      @cost = input[:cost]
-      @rating = input[:rating]
+      @start_time = check_start_time(input[:start_time])
+      @end_time = check_end_time(input[:end_time])
+      @cost = check_cost(input[:cost])
+      @rating = check_rating(input[:rating])
 
-      if @rating > 5 || @rating < 1
-        raise ArgumentError.new("Invalid rating #{@rating}")
+      check_time
+
+    end
+
+    def duration
+      @end_time - @start_time
+    end
+
+    def inspect
+      "#<#{self.class.name}:0x#{self.object_id.to_s(16)}>"
+    end
+
+    private
+
+    def check_cost(cost_input)
```

Comment:
Ruby 2.x has a funny new syntax where you can basically collapse this `nil` checking down to a one-liner:

```ruby
def check_cost(cost)
  cost&.to_f
end
```

The `&.` syntax basically means "if the object is nil, don't call the method, otherwise call it if we can"

Reply:
Rails historically had/has a similar method called `try()`, where you would do:

```
cost.try(:to_f)
```

and it would try to call the function if it could